samples/boards/tlsr9x/timer/src/main.c

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2024 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <timer.h>
+#include <zephyr/irq.h>
+#include <string.h>
+
+#define TIMER_MODE_GPIO_TRIGGER_TICK    0xA
+
+#define TIMER_CAPT_1            GPIO_PA0
```

Comment:
HW should be provided from .dts

Reply:
For this @BorysNykytiuk created separate Jira ticket for @mishadesh http://192.168.48.49:8080/browse/CS-1563
In this case maybe better to postpone merge till this ticket is not finished

Reply:
Totaly agree

Reply:
PR is ready for review

Reply:
PR is ready for review